### PR TITLE
limit env's to only selected projects, or my projects

### PR DESCRIPTION
**Pull request by nhsiehgit**
Currently we list _every_ environment if you are a superuser.

This PR modifies the filter to only present environments for the selected project, or "my projects" if selected

### CODE REVIEW

File: static/app/views/releases/thresholdsList/index.tsx [hunk 1]
```diff
@@ -112,6 +112,39 @@ function ReleaseThresholdList({}: Props) {
...
+
+        return [];
+      })
+    );
+    const envDiff = new Set([...allEnvSet].filter(x => !unSortedEnvs.has(x)));
+
+    // bubble the selected projects envs first, then concat the rest of the envs
+    return Array.from(unSortedEnvs)
+      .sort()
+      .concat([...envDiff].sort());
```
Comment (saponifi3d) [inline static/app/views/releases/thresholdsList/index.tsx:145]: Looks like this copying and sorting the `envDiff` but doesn't seem like we need to make a copy
```suggestion
      .concat(envDiff.sort());
```

File: static/app/views/releases/thresholdsList/index.tsx [hunk 2]
```diff
@@ -112,6 +112,39 @@ function ReleaseThresholdList({}: Props) {
...
+          (selectedProjectIds.length === 0 && project.isMember) ||
+          selectedProjectIds.includes(project.id)
+        ) {
+          return project.environments;
+        }
+
+        return [];
+      })
+    );
+    const envDiff = new Set([...allEnvSet].filter(x => !unSortedEnvs.has(x)));
```
Comment (saponifi3d) [inline static/app/views/releases/thresholdsList/index.tsx:140]:
```suggestion
    const envDiff = new Set(allEnvSet.filter(x => !unSortedEnvs.has(x)));
```

**Review by saponifi3d — APPROVED** [review]
🎉

File: static/app/views/releases/thresholdsList/index.tsx (hunk 1 above)
Comment (nhsiehgit) [reply static/app/views/releases/thresholdsList/index.tsx:145]: Ah - the `envDiff` is a set, so this is me splatting that into an array and sorting alphabetically

**Pull request merged**